### DISCUSSION TITLE
include does not resolve `~`

### DISCUSSION THREAD

**Discussion by papanito** [twpayne/chezmoi#2263] [Q&A] [answered]
I would like to use `include` as follows

```
{{- if eq .chezmoi.hostname "myhost" -}}
{{ include "~/.kube/config.prd" }}
{{ else }}
{{ include "~/.kube/config.nop" }}
{{ end }}
```

However it seems, `~` is not resolved

```
chezmoi: template: dot_kube/config.tmpl:4:3: executing "dot_kube/config.tmpl" at <include "~/.kube/config.nop">: error calling include: open C:/Users/papanito/.local/share/chezmoi/~/.kube/config.nop: The system cannot find the path specified.
```

It works if I do this

```
{{- if eq .chezmoi.hostname "myhost" -}}
{{ include "C:/Users/papanito/.kube/config.prd" }}
{{ else }}
{{ include "C:/Users/papanito/.kube/config.nop" }}
{{ end }}
```

Am I doing something wrong or is that a bug?

**Comment by twpayne** [comment, marked as answer]
The `~` character is a valid character in filenames. Many shells, however, do tilde expansion and interpret it as a shortcut for `$HOME` when interpreting the arguments to a command. Tilde expansion is the responsibility of the shell, not chezmoi. For more information, see #952.

The portable workaround is to use `.chezmoi.homeDir`, for example:

```
{{ include (joinPath .chezmoi.homeDir ".kube" "config.prd") }}
```
